sprint0/sprint0/Weapons/Fireball.cs

```diff
@@ -14,32 +14,41 @@ public class Fireball : IProjectile
         public Texture2D Texture { get; set; }
         public int Damage { get => 1; }
         private readonly int width = 8, height = 10;
-        private readonly List<Rectangle> sources;
+        private readonly Dictionary<Color, List<Rectangle>> colorMap;
+        private readonly Color color;
         private int currFrame;
         private readonly int totalFrames, repeatedFrames, speed = 3;
         private readonly Vector2 direction;
         private bool hit = false;
 
-        public Fireball(Texture2D texture, Vector2 location, Vector2 direction, IEntity shooter)
+        public Fireball(Texture2D texture, Vector2 location, Vector2 direction, IEntity shooter, Color color)
         {
             Shooter = shooter;
             Texture = texture;
+            this.color = color;
             Location = new Rectangle((int)location.X, (int)location.Y, (int)(width * Game1.Scale), (int)(height * Game1.Scale));
             preciseLocation = location;
             this.direction = direction;
 
             currFrame = 0;
-            totalFrames = 4;
+            totalFrames = 1;
             repeatedFrames = 2;
-            sources = SpritesheetHelper.GetFramesH(231, 62, width, height, totalFrames);
+            colorMap = new Dictionary<Color, List<Rectangle>>
+            {
+                {Color.Yellow, SpritesheetHelper.GetFramesH(231, 62, width, height, totalFrames)},
+                {Color.Blue, SpritesheetHelper.GetFramesH(240, 62, width, height, totalFrames)},
+                {Color.Red, SpritesheetHelper.GetFramesH(249, 62, width, height, totalFrames)},
+                {Color.Green, SpritesheetHelper.GetFramesH(258, 62, width, height, totalFrames)},
+            };
+                
```

Comment:
I think you could just use sprite sheet helper as it was originally and the make a dictionary from the list the sprite sheet helper returns